packages/react-core/src/components/ProgressStepper/ProgressStep.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+import styles from '@patternfly/react-styles/css/components/ProgressStepper/progress-stepper';
+import { css } from '@patternfly/react-styles';
+import CheckCircleIcon from '@patternfly/react-icons/dist/esm/icons/check-circle-icon';
+import ResourcesFullIcon from '@patternfly/react-icons/dist/esm/icons/resources-full-icon';
+import ExclamationTriangleIcon from '@patternfly/react-icons/dist/esm/icons/exclamation-triangle-icon';
+import ExclamationCircleIcon from '@patternfly/react-icons/dist/esm/icons/exclamation-circle-icon';
+
+export interface ProgressStepProps
+  extends React.DetailedHTMLProps<React.LiHTMLAttributes<HTMLLIElement>, HTMLLIElement> {
+  /** Content rendered inside the progress step. */
+  children?: React.ReactNode;
+  /** Additional classes applied to the progress step container. */
+  className?: string;
+  /** Variant of the progress step. Each variant has a default icon. */
+  variant?: 'default' | 'success' | 'info' | 'pending' | 'warning' | 'danger';
+  /** Flag indicating the progress step is the current step. */
+  isCurrent?: boolean;
+  /** Custom icon of a progress step. Will override default icons provided by the variant. */
+  icon?: React.ReactNode;
+  /** Description text of a progress step. */
+  description?: string;
+  /** ID of the title of the progress step. */
+  titleId?: string;
+}
+
+const variantIcons = {
+  default: undefined as any,
+  pending: undefined as any,
+  success: <CheckCircleIcon />,
+  info: <ResourcesFullIcon />,
+  warning: <ExclamationTriangleIcon />,
+  danger: <ExclamationCircleIcon />
+};
+
+const variantStyle = {
+  default: '',
+  info: styles.modifiers.info,
+  success: styles.modifiers.success,
+  pending: styles.modifiers.pending,
+  warning: styles.modifiers.warning,
+  danger: styles.modifiers.danger
+};
+
+export const ProgressStep: React.FunctionComponent<ProgressStepProps> = ({
+  children,
+  className,
+  variant,
+  isCurrent,
+  description,
+  icon,
+  titleId,
+  ...props
+}: ProgressStepProps) => {
+  const _icon = icon !== undefined ? icon : variantIcons[variant];
+  return (
+    <li
+      className={css(
+        styles.progressStepperStep,
+        variantStyle[variant],
+        isCurrent && styles.modifiers.current,
+        className
+      )}
+      {...props}
+    >
+      <div className={css(styles.progressStepperStepConnector)}>
+        <span className={css(styles.progressStepperStepIcon)}>{_icon && _icon}</span>
+      </div>
+      <div className={css(styles.progressStepperStepMain)}>
+        <div
+          className={css(styles.progressStepperStepTitle)}
+          id={titleId}
+          {...(props.id !== undefined && titleId !== undefined && { 'aria-labelledby': `${props.id} ${titleId}` })}
```

Comment:
Should the `aria-labeled` be on the `<li> since that is what is being labeled? `id` should be added there as well if that is the case. 
Should we generate the id her if non is supplied? If we don't, will this be accessible?  If so, we should probably generate warning if step is missing a title since it will not be accessible.

Reply:
So my concern with putting the `aria-labelledby` on the li is that I don't think the screen reader really shifts focus there consistently. The visible text will get focus every time which made me think it would be more consistent to put it there. I did notice that at times the li's `aria-label` wasn't always announced (depending on the screen reader) so we could investigate further the best way to label these. In terms of the title though, I do think a warning could be valuable.

Reply:
What do you mean by warning?

Reply:
@kmcfaul  similar to what we do with other components.  Log a console warning if the props required to make the component accessible  are missing.

Reply:
Added a console warning for missing either `id` or `titleId` (which are used to create the `aria-labelledby`).